### ISSUE TITLE
Deploy Galaxy with Ansible

### ISSUE THREAD

**Issue by dtitov** [elixir-oslo/oncoEnrichR_galaxy#5]

**Comment by dtitov** [comment]
Resources on Galaxy admin/setup:

- Galaxy Github repo: https://github.com/galaxyproject/galaxy
    - In order to build local offline documentation (e.g. as in https://docs.galaxyproject.org/en/master/):
        - Clone Github repo locally
        - Run "make docs" 
- https://github.com/galaxyproject/training-material (Can be installed locally. Includes slides on installing galaxy with Ansible)
- Ansible playbook for Galaxy: https://github.com/galaxyproject/usegalaxy-playbook/
- Galaxy Hub Website repo: https://github.com/galaxyproject/galaxy-hub

**Comment by sveinugu** [comment]
@nemyxun Great work with setting up the server!

Copying in information from your email:

> [centos@oncoenrich-server usegalaxy-playbook]$ ansible-galaxy list
> # /home/centos/usegalaxy-playbook/roles
> - galaxyproject.interactive_environments, 0.4
> - galaxyproject.proftpd, 0.2.2
> - natefoo.postgresql_objects, 1.1
> - galaxyproject.postgresql, 1.0.2
> - galaxyproject.galaxy, 0.9.2
> - galaxyproject.galaxy_toolshed, 0.2.2
> - usegalaxy-eu.certbot, (unknown version)
> - galaxyproject.nginx, 0.6.4
> - galaxyproject.pulsar, 1.0.4
> # /home/centos/usegalaxy-playbook/common_roles

Looking at the above-mentioned list, I have some additional comments/tasks:

- [ ] Interactive environments need Docker to run. We will skip it for now, but want to add it at a later stage, so I suppose the playbook can stay in if it does not cause trouble. We will revisit this when we have a secure solution for running Docker.
- [ ] The Galaxy Tool Shed is not needed, so you can remove the playbook. This is for publishing our own Galaxy tools internally. We already have a Elixir Norway tool shed running in Trondheim (NTNU), and we can use that if we need to publish any tools internally, we do not need a new server. In the current case, I think the tool anyway be should be published to the public tool shed: https://toolshed.g2.bx.psu.edu/.
- [ ] Pulsar is for offloading jobs to other Galaxy worker nodes. We do not need this now, but I suppose it does not hurt to have it installed if we might need it later.
- [ ] The Certbot (for Let's Encrypt) is useful. However, to make is work, we need a domain. See comment below about finding a domain name. Once we have that, you can install Let's Encrypt automatic certificates.
- [ ] There are certain configurations to be done to set up Galaxy in a production environment: https://docs.galaxyproject.org/en/master/admin/production.html. Many of these are set up by default by the Ansible script, I presume, but all might not be. I suggest you go through these step by step and double-check that they have been set up, or if not, set them up. Note any changes done so that this can be automated later. This includes:
- [ ] Debug modes are turned off
- [ ] A secret key has been set for id_secret (https://docs.galaxyproject.org/en/latest/admin/config.html#id-secret). For some reason I could not find any info in this in the production setup page, which I find very strange. I'm sure it is documented somewhere.
- [ ] Basic log rotation
- [ ] Setting up a FTP server to be used upload large files
- [ ] Set up load balancing on the web site (At least 2 web servers)
- [ ] Postgres optimal database settings
- [ ] Add admin users (in this case: sveinugu@ifi.uio.no, radmilko@ifi.uio.no, and yourself)
- [ ] Try to install a tool from the main too shed with Conda dependencies and test that the Conda dependencies are correctly installed.
- [ ] There is an bug reporting mechanism in Galaxy that can send an email. https://docs.galaxyproject.org/en/master/admin/special_topics/bug_reports.html. It requires a running SMTP server in the server. See https://docs.galaxyproject.org/en/master/admin/special_topics/bug_reports.html. Relevant galaxy.yaml config lines: https://docs.galaxyproject.org/en/master/admin/config.html#smtp-server (value: "localhost"), https://docs.galaxyproject.org/en/master/admin/config.html#error-email-to (value: "hyperbrowserbugreport+al0ybw2kfsws1t0jfyxo@boards.trello.com").

Also, some reporting:

- [ ] Could you provide an overview of how the is the users and services are set up in the Galaxy Ansible setup. I assume there is a galaxy user and a postgres user, in addition to the standard login users. Is there a sudo user setup so that the login user (centos?) can sudo into the other two users, as well as root (which is the setup we have in our previous manual installations of Galaxy servers)?
- [ ] Also, I suppose there are no general command-line Conda support in the Ansible script, only the Conda client that is installed within the Galaxy directory ("database/_conda/bin" I believe, or similar). Could you query Nicolay and find out how he has installed a general conda client and provide info to @sveinugu? Then we can consider whether/how we would like to support a command-line conda installation.

**Comment by sveinugu** [comment]
@sigven @dtitov @ehovig Any ideas for the domain name? oncoenrichr.elixir.no? Seems not so readable/rememberable to me. What about just onco.elixir.no or oncotools.elixir.no? Once we have the name, I can send an email to set up the domain.

**Comment by dtitov** [comment]
@sveinugu, I don't really have any strong preferences. 

I would personally go for the `oncoenrichr.elixir.no` because it maps one-to-one to the name of the project (although, I thought that the name of the project is `oncoEnrich`, without `R` at the end - good to know). 

`onco.elixir.no` or `oncotools.elixir.no` sounds very broad to me, like there's more than one tool there, which is not true. At least so far, before we add something else. Are there any plans, BTW, to add some more onco-related tools to this server in the future? If so, then, I think `oncotools.elixir.no` is a good option (`onco.elixir.no` is still way too vague IMHO).

**Comment by nemyxun** [comment]
I think we can use multiply names
onco.elixir.no is so much preferable for me

**Comment by sveinugu** [comment]
I think @sigven and @ehovig should have the last word regarding the name.

**Comment by sigven** [comment]
Hi,
@dtitov raises a good point. Although this project is all about supporting one tool, I believe it's smart to not completely rule out the possibility of adding more related tools in the future. In that case I think `oncotools.elixir.no` is the best suggestion.

**Comment by dtitov** [comment]
Ok, sounds good. Should I request `oncotools.elixir.no` or do we wait for @ehovig?

**Comment by ehovig** [comment]
I agree with Sigve here. He may have more stuff up his sleeve... B-)

 

Eivind

 

Fra: Sigve Nakken <notifications@github.com>
Svar til: elixir-oslo/oncoEnrichR_galaxy <reply@reply.github.com>
Dato: onsdag 8. april 2020 15:34
Til: elixir-oslo/oncoEnrichR_galaxy <oncoEnrichR_galaxy@noreply.github.com>
Kopi: Eivind Hovig <ehovig@ifi.uio.no>, Mention <mention@noreply.github.com>
Emne: Re: [elixir-oslo/oncoEnrichR_galaxy] Deploy Galaxy with Ansible (#5)

 

Hi,
@dtitov raises a good point. Although this project is all about supporting one tool, I believe it's smart to not completely rule out the possibility of adding more related tools in the future. In that case I think oncotools.elixir.no is the best suggestion.

**Comment by ehovig** [comment]
No need to wait!

Eivind

 

Fra: Dmytro Titov <notifications@github.com>
Svar til: elixir-oslo/oncoEnrichR_galaxy <reply@reply.github.com>
Dato: onsdag 8. april 2020 16:06
Til: elixir-oslo/oncoEnrichR_galaxy <oncoEnrichR_galaxy@noreply.github.com>
Kopi: Eivind Hovig <ehovig@ifi.uio.no>, Mention <mention@noreply.github.com>
Emne: Re: [elixir-oslo/oncoEnrichR_galaxy] Deploy Galaxy with Ansible (#5)

 

Ok, sounds good. Should I request oncotools.elixir.no or do we wait for @ehovig?

**Comment by dtitov** [comment]
Thanks, Eivind. Ok, I'm requesting `oncotools.elixir.no`.

**Comment by nemyxun** [comment]
Hello All,
Quick report. PLZ let me know if something's wrong.
Docker CE has been installed from official Docker CE Stable repo.
Certbot has been installed from EPEL repo.
Let's Encrypt has been installed from GitHub.
Awaiting for DNS record for new certs generation, Mail Server setup and etc.
THX

**Comment by sveinugu** [comment]
Hi Lev, Regarding Docker, I think I mentioned that we can postpone that, as we need to figure out a secure way of running with Docker. There is work done on this in the tech department at UiO (USIT), as well as in Elixir Bergen, so a bit of coordination is needed. As Docker in itself is a security risk, I propose you uninstall it for now..

As to Certbot, I assumed that the Ansible playbook you installed would contain everything, so that there were no need to install anything else other than configure the installation. But I might be wrong.

As to the SMTP server, you requested some additions. Can you explain what these are for? I don't think we have needed any changes to the DNS before, but I am no expert on these things. 

Anyway, please be aware that today, Friday and Monday are public holidays in Norway, so nothing is expected from you these days (and we might not respond).

**Comment by nemyxun** [comment]
Hello All
Galaxy and FTP services have been stopped during long Easter holiday.
I've asked to add DKIM record because messages may be marked as spam or may be refused by some mail servers.
For example, my Gmail refuse all mail from untrusted source e.g. without PTR and DKIM and etc.

Pure-FTP server has been installed from CentOS repo.
Certbot does not work without DNS record.
Awaiting for.
THX

**Comment by nemyxun** [comment]
Hello All
Galaxy log files has been organized to /var/log/galaxy
1st one - daemon started log, 2nd - access log.
Attached to logrotate daemon for weekly processing.
Any advise how long we have to keep them up?
THX
